cfme/services/myservice.py

```diff
@@ -148,15 +216,16 @@ def retire_on_date(self, retirement_date):
         assert vm_state in ['off', 'unknown']
 
     def update(self, updates):
-        navigate_to(self, 'Edit')
-        updated_name = updates.get('service_name', self.service_name + '_edited')
-        updated_description = updates.get('description', 'Updated description')
-        fill(edit_service_form, {'name': updated_name,
-                                 'description': updated_description},
-             action=form_buttons.angular_save)
-        wait_for(flash.get_messages, timeout=10, delay=2, fail_condition=[], fail_func=tb.refresh())
-        if flash.assert_success_message('Service "{}" was saved'.format(updated_name)):
-            setattr(self, 'service_name', updated_name)
+        view = navigate_to(self, 'Edit')
+        changed = view.fill(updates)
+        if changed:
+            view.save_button.click()
+        else:
+            view.cancel_button.click()
+        view = self.create_view(MyServiceDetailView, override=updates)
+        assert view.is_displayed
+        view.flash.assert_success_message('Service "{}" was saved'.
+            format(updates.get('name', self.name)))
```

Comment:
When you are handling the cancel button in this method, you should also change the respective flash message checking. Applies to all methods where this happens.